src/tck/Reactive.Streams.TCK.Tests/PublisherVerificationTest.cs

```diff
@@ -0,0 +1,637 @@
+            });
+            var verification = CustomPublisherVerification(publisher);
+
+            RequireTestFailure(() => verification.Required_spec105_mustSignalOnCompleteWhenFiniteStreamTerminates(),
+                messagePart: "Expected end-of-stream but got element [3]");
+        }
+
+        [Test]
+        public void Optional_spec105_emptyStreamMustTerminateBySignallingOnComplete_shouldNotAllowEagerOnComplete()
+        {
+            var publisher = new LamdaPublisher<int>(onSubscribe: subscriber => subscriber.OnComplete());
+            var verification = new Spec105Verification(NewTestEnvironment(), publisher);
+            RequireTestFailure(() => verification.Optional_spec105_emptyStreamMustTerminateBySignallingOnComplete(),
+                "Subscriber.OnComplete() called before Subscriber.OnSubscribe");
+        }
+
+        [TestFixture(Ignore = "Helper for single test")]
+        private sealed class Spec105Verification : PublisherVerification<int>
+        {
+            private readonly IPublisher<int> _publisher;
+
+            /// <summary>
+            /// We need this constructor for NUnit even if the fixture is ignored 
+            /// </summary>
+            public Spec105Verification() : base(NewTestEnvironment())
+            {
+                
+            }
+
+            public Spec105Verification(TestEnvironment environment, IPublisher<int> publisher) : base(environment)
+            {
+                _publisher = publisher;
+            }
+
+            public override IPublisher<int> CreatePublisher(long elements) => _publisher;
+
+            public override IPublisher<int> CreateFailedPublisher() => null;
+
+            public override long MaxElementsFromPublisher { get; } = 0; // it is an "empty" Publisher
+        }
+
+        [Test]
+        public void Required_spec107_mustNotEmitFurtherSignalsOnceOnCompleteHasBeenSignalled_shouldFailForNotCompletingPublisher()
+            => RequireTestFailure(() => DemandIgnoringAsynchronousPublisherVerification().Required_spec107_mustNotEmitFurtherSignalsOnceOnCompleteHasBeenSignalled(),
+                "Expected end-of-stream but got element ["/* element which should not have been signalled */);
+
+        [Test]
+        public void Required_spec107_mustNotEmitFurtherSignalsOnceOnCompleteHasBeenSignalled_shouldFailForPublisherWhichCompletesButKeepsServingData()
+        {
+            var publisher = new LamdaPublisher<int>(onSubscribe: subscriber =>
+            {
+                var completed = false;
+
+                subscriber.OnSubscribe(new LamdaSubscription(onRequest: n =>
+                {
+                    // emit one element
+                    subscriber.OnNext(0);
+
+                    // and "complete"
+                    // but keep signalling data if more demand comes in anyway!
+                    if (!completed)
+                    {
+                        subscriber.OnComplete();
+                        completed = true;
```

Comment:
Interesting :) I think I haven't noticed that it's not there in the jvm, because if so I have had said something ;)